public/js/applicationForm.js

```diff
@@ -0,0 +1,129 @@
+const navButton = document.getElementById('applicationsNav');
+navButton.classList.add('activePage');
+
+const params = new URLSearchParams(window.location.search);
+const applicationId = params.get('_id');
+
+const positionInput = document.getElementById('position');
+const postulantInput = document.getElementById('postulant');
+const interviewInput = document.getElementById('interview');
+const resultInput = document.getElementById('result');
+
+const form = document.getElementById('form');
+const saveButton = document.getElementById('saveButton');
+const errorMessage = document.getElementById('error_message');
+
+saveButton.disabled = !!applicationId;
+
+const onFocusInput = () => {
+  errorMessage.innerText = '';
+};
+
+resultInput.onfocus = onFocusInput;
+
+const fillSelect = (url, parent) => {
+  fetch(`${window.location.origin}/api${url}`)
+    .then((response) => {
+      if (response.status !== 200) {
+        return response.json().then(({ message }) => {
+          throw new Error(message);
+        });
+      }
+      return response.json();
+    })
+    .then((response) => {
+      saveButton.disabled = false;
+      response.data.forEach((el) => {
+        const option = document.createElement('option');
+        // eslint-disable-next-line no-underscore-dangle
+        option.value = el._id;
+        option.innerText = `${el.firstName ? el.firstName : ''} ${el.firstName ? el.lastName : ''}
+        ${el.jobDescription ? el.jobDescription : ''}
+        ${el.date ? el.date.slice(0, 19).replace('T', ' ') : ''}`;
+        parent.append(option);
+      });
+    })
+    .catch((error) => {
+      errorMessage.innerText = error;
+    });
+};
+
+fillSelect('/positions', positionInput);
+fillSelect('/postulants', postulantInput);
+fillSelect('/interviews', interviewInput);
```

Comment:
You are creating a reusable function but part of the logic is tied to the possible implementations.

```suggestion
const fillSelect = (url, parent, makeText) => {
  fetch(`${window.location.origin}/api${url}`)
    .then((response) => {
      if (response.status !== 200) {
        return response.json().then(({ message }) => {
          throw new Error(message);
        });
      }
      return response.json();
    })
    .then((response) => {
      saveButton.disabled = false;
      response.data.forEach((el) => {
        const option = document.createElement('option');
        // eslint-disable-next-line no-underscore-dangle
        option.value = el._id;
        option.innerText = makeText(el);
        parent.append(option);
      });
    })
    .catch((error) => {
      errorMessage.innerText = error;
    });
};

fillSelect('/positions', positionInput, (el) => `${el.firstName} ${el.lastName}`);
fillSelect('/postulants', postulantInput, (el) => el.jobDescription);
fillSelect('/interviews', interviewInput, (el) => el.date.slice(0, 19).replace('T', ' '));
```